tests/conformance/media_key_system_access.html

```diff
@@ -0,0 +1,179 @@
+<!DOCTYPE html>
+<!--
+
+This file allows to quickly test which MediaKeySystemAccess is supported on the
+current device.
+
+To do that:
+
+  1. update the `configs` property in the script part of the document.
+
+  2. Run this page from the environment (device and/or browser) you want to test
+
+  3. Check the logs, that's it!
+
+-->
+<html lang="en">
+  <head>
+    <head>
+      <meta charset="UTF-8">
+      <title>RxPlayer Conformance Test - MediaKeySystemAccess support</title>
+    </head>
+    <body>
+      <script charset="utf-8">
+/* eslint-disable no-console */
+
+/* Edit this variable to test available MediaKeySystemAccess configurations. */
+const configs = [
+  {
+    keySystem: "com.widevine.alpha",
+    configuration: [{
+      initDataTypes: ["cenc"],
+      audioCapabilities:[{
+        robustness: "",
+        contentType: "audio/mp4; codecs=\"mp4a.40.2\"",
+      }],
+      videoCapabilities:[{
+        robustness: "",
+        contentType: "video/mp4;codecs=\"avc1.640028\"",
+      }],
+      sessionTypes: ["temporary"],
+    }],
+  },
+  {
+    keySystem: "com.microsoft.playready",
+    configuration: [{
+      initDataTypes: ["cenc"],
+      audioCapabilities:[{
+        robustness: "",
+        contentType: "audio/mp4; codecs=\"mp4a.40.2\"",
+      }],
+      videoCapabilities:[{
+        robustness: "",
+        contentType: "video/mp4;codecs=\"avc1.640028\"",
+      }],
+      sessionTypes: ["temporary"],
+    }],
+  },
+];
+
+start();
+
+function start() {
+  recursivelyTestConfig(0);
+}
+
+/**
+ * Recursively tests all MediaKeySystemConfiguration configured in the `configs`
+ * object from the given `idx` index.
+ * @param {number} idx
+ */
+function recursivelyTestConfig(idx) {
+  if (idx >= configs.length) {
+    console.log("Done!");
+    return;
+  }
+
+  const videoElement = createVideoElement();
+  const { keySystem, configuration } = configs[idx];
+  console.log("Testing new config:", keySystem, configuration);
+  createMediaKeySystemAccess(keySystem, configuration)
+    .then((mediaKeySystemAccess) => {
+      return createMediaKeys(mediaKeySystemAccess);
+    })
+    .then((mediaKeys) => {
+      return attachMediaKeysToVideoElement(mediaKeys, videoElement);
+    }).then(() => {
+      console.log(`Configuration ${idx} succeeded!`,
+                  keySystem,
+                  configuration);
+    }).finally(() => {
```

Comment:
Maybe you could catch here a potential error.
It seems that when a keySystem is rejected, you throw, doesn't catch and it interrupts the flow.

Reply:
I don't see any case where an error is not logged.
It is done in `createMediaKeySystemAccess`, `createMediaKeys` and `attachMediaKeysToVideoElement`, no?

Reply:
Yes but do we really want to throw in those functions ?

Reply:
I don't get what you mean. In any case `finally` is going to catch everything and just test the next config, no?